### ISSUE TITLE
Validation checks only the explicit rules not the interpretation

### ISSUE THREAD

**Issue by nickerso** [cellml/libcellml#261]
Highlighted by #260, the current validation performed by libCellML checks only the explicit rules in the specification but not the rules that are part of the interpretation of the specification. We need to decide whether to include those rules as part of the validation or if this is an indication that such rules need to be explicitly stated in the specification.

**Comment by agarny** [comment]
I quite like the current format of the [CellML 2.0 specification](https://www.cellml.org/specifications/cellml_2.0), so I would probably leave it  as-is, not least since it would otherwise break rule 2.6.2. This also means that we might indeed need a special validation document?

**Comment by nickerso** [comment]
I think the rules are all in the spec, we might just want to add "interpretation" validation to match the syntactical validation that we currently have.

**Comment by agarny** [comment]
Wouldn't that "break" the normative aspect of the specification (see rule 2.6.2)? If not, then I am happy with that.

**Comment by nickerso** [comment]
I don't think so, the interpretation section (19) is already part of the normative specification and we use parts of it already (i.e., the built in units). Not adding anything to the CellML infoset simply extending the rules that a valid model document must obey.

**Comment by agarny** [comment]
Oops, I did miss/forget about that section. Ok, all fine with it then.
